tmd/view/view.py

```diff
@@ -1115,10 +1113,166 @@ def polar_plot(pop, neurite_type="neurites", bins=20):
+        return None, "No such plane found! Please select one of: xy, xz, yx, yz, zx, zy."
+
+    # Initialization of matplotlib figure and axes.
+    fig, ax = cm.get_figure(new_fig=new_fig, new_axes=new_axes, subplot=subplot)
+
+    # Initialization of colors to be used.
+    if colors is None:
+        ordered_nums = [np.float(i) / len(ph_graph) for i in np.arange(len(ph_graph))]
+        colors_random = cmap(ordered_nums)
+    else:
+        colors_random = colors
+
+    # Definition of tree branch colors based on persistence
+    def get_colors_ph(tree, ph_graph, colors):
+        """Assigns colors to each tree branch according to the persistence levels."""
+        beg, end = tree.get_sections_2()
+        end_graph = {}
+        for j, graph_id in enumerate(ph_graph):
+            for gi in graph_id:
+                end_id = np.where(end == gi)[0][0]
+                end_graph[end_id] = j
+
+        sec_ids = [1] * (end[0] - beg[0])
+        for i, e in enumerate(end[1:]):
+            sec_ids = sec_ids + [i + 2] * (e - end[i])
+
+        colors_select = [colors[end_graph[s - 1]] for s in sec_ids]
+        return colors_select
+
+    treecolors = get_colors_ph(tree, ph_graph, colors_random)
+
+    # Data needed for the viewer: x,y,z,r
+    bounding_box = tree.get_bounding_box()
+
+    def _seg_2d(seg):
+        """2d coordinates required for the plotting of a segment."""
+        horz = term_dict[plane[0]]
+        vert = term_dict[plane[1]]
+
+        horz1 = seg[0][horz]
+        horz2 = seg[1][horz]
+        vert1 = seg[0][vert]
+        vert2 = seg[1][vert]
+
+        return ((horz1, vert1), (horz2, vert2))
+
+    segs = [_seg_2d(seg) for seg in tree.get_segments()]
+
+    # Definition of the linewidth according to diameter, if diameter is True.
+
+    linewidth = [d * 1.0 for d in tree.d]
+
+    # Plot the collection of lines.
+    collection = _LC(segs, color=treecolors, linewidth=linewidth, alpha=1.0)
+
+    ax.add_collection(collection)
+
+    ax.set_xlim(bounding_box[0][term_dict[plane[0]]], bounding_box[1][term_dict[plane[0]]])
+    ax.set_ylim(bounding_box[0][term_dict[plane[1]]], bounding_box[1][term_dict[plane[1]]])
+
+    kwargs["title"] = kwargs.get("title", "Tree structure")
+    kwargs["xlabel"] = kwargs.get("xlabel", plane[0])
+    kwargs["ylabel"] = kwargs.get("ylabel", plane[1])
+
+    white_space = _get_default("white_space", **kwargs)
+    kwargs["xlim"] = kwargs.get(
+        "xlim",
+        [
+            bounding_box[0][term_dict[plane[0]]] - white_space,
+            bounding_box[1][term_dict[plane[0]]] + white_space,
+        ],
+    )
+    kwargs["ylim"] = kwargs.get(
+        "ylim",
+        [
+            bounding_box[0][term_dict[plane[1]]] - white_space,
+            bounding_box[1][term_dict[plane[1]]] + white_space,
+        ],
+    )
+
+    return cm.plot_style(fig=fig, ax=ax, **kwargs)
+
+
+def tree_barcode_colors(tree, plane="xy", feature="path_distances", cmap=cm.jet_map):
```

Comment:
The `plane` parameter is not used, is it expected?

Reply:
correct, I used the intermediate plot to color the tree, but it should be passed down to the internal function.